shared/macros/10-oval.jinja

```diff
@@ -1071,19 +1072,30 @@ Generates the :code:`<affected>` tag for OVAL check using correct product platfo
         <extend_definition comment="rpm package openssh-server installed"
           definition_ref="package_openssh-server_installed" />
         {{% endif %}}
-        <criteria comment="sshd is configured correctly" operator="AND">
-          <criteria comment="the configuration is correct if it exists" operator="AND">
-            {{{- oval_line_in_file_criterion(sshd_config_path, parameter, avoid_conflicting=true, rule_id=rule_id) | indent(10)}}}
-            {{%- if config_is_distributed == "true" %}}
-            {{{- oval_line_in_directory_criterion(sshd_config_dir, parameter, avoid_conflicting=true, rule_id=rule_id) | indent(10) }}}
+        <criteria comment="sshd is configured correctly" operator="OR">
+          {{%- if runtime_check == "true" %}}
+          <criteria comment="runtime configuration exists and is correct" operator="AND">
+            <criterion comment="runtime config file exists" test_ref="test_runtime_config_present_{{{ rule_id }}}" />
+            <criterion comment="runtime config matches expected value" test_ref="test_runtime_{{{ parameter }}}_{{{ rule_id }}}" />
+          </criteria>
+          {{%- endif %}}
+          <criteria comment="static configuration is correct (when runtime config doesn't exist or runtime check disabled)" operator="AND">
+            {{%- if runtime_check == "true" %}}
+            <criterion comment="runtime config file does not exist" test_ref="test_runtime_config_absent_{{{ rule_id }}}" />
             {{%- endif %}}
```

Comment:
If `runtime_check` is true we should enforce the run time settings are checked.

It could be confusing that we set RHCOS4 to check SSHD runtime settings, but evaluate to pass simply because the file is missing. If the file is missing and `runtime_check == true`, IMHO, the rule should fail.